### PR TITLE
ConstellationColor Unit Tests

### PR DESCRIPTION
**Pull request by antares1470**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first.

### Description of the Change

Wrote unit tests for the `ConstellationColor` class. As I was writing them, I discovered a few issues with the class itself, namely a few missing null checks and broken logic in the `compareTo` function so that has been fixed.

### Why Should This Be In Core?

Unit tests are good

### Benefits

More unit tests

### Applicable Issues

#413

### CODE REVIEW

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java [hunk 1]
```diff
@@ -463,10 +502,31 @@ public int compareTo(final ConstellationColor o) {
...
+     * @return integer representing the result of the comparison
+     */
+    private int compareColourComponents(final float colour1Component, final float colour2Component) {
+        final float comparison = colour1Component - colour2Component;
+        if (comparison > 0) {
+            return 1;
+        } else if (comparison < 0) {
+            return -1;
+        } else {
+            return 0;
```
[the remark this answers is not in the document]
Comment (antares1470) [reply CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java]: Done

**Review by arcturus2 — APPROVED** [review]

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java [hunk 2]
```diff
@@ -424,6 +432,37 @@ public static ConstellationColor fromRgbWithCommaColor(final String color) {
...
+     * actual size may be less.
+     *
+     * @param s The string to split.
+     * @param size The expected size of the resulting array.
+     * @param separator The separator character to split on.
+     *
+     * @return A String[] containing the fields that have been split out of the
+     * string.
+     */
+    private static String[] split(final String s, final int size, final char separator) {
```
Comment (aldebaran30701) [inline CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:449]: Is this not better suited to the `StringUtilities` class?
Noted that it was always here.

**Review by aldebaran30701 — COMMENTED** [review]
Looks great!

File: CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java (hunk 2 above)
Comment (antares1470) [reply CoreUtilities/src/au/gov/asd/tac/constellation/utilities/color/ConstellationColor.java:449]: The answer is probably (though I think that can be saved for another time 😄)

**Pull request merged**